VisualizationBase/src/ViewItemManager.cpp

```diff
@@ -121,6 +124,57 @@ void ViewItemManager::removeAllViewItems()
 	currentViewItem_ = nullptr;
 }
 
+ViewItem* ViewItemManager::createIfNotExists(const QString name, QPoint position)
+{
+	//Try seeing if it is opened already
+	if (auto view = viewItem(name))
+		return view;
+	//Else check if we have it persisted somewhere
+	for (auto manager : Model::AllTreeManagers::instance().loadedManagers())
+		if (auto view = loadView(name, manager))
+		{
+			addViewItem(view, position);
+			return view;
+		}
+	//If it doesn't exist, create a new view item
+	return newViewItem(name, position);
+}
+
+void ViewItemManager::saveView(ViewItem* view, Model::TreeManager* manager) const
+{
+	auto json = view->toJson().toJson();
+	QFile file(fileName(view->name(), manager->name()));
+	file.open(QIODevice::WriteOnly);
+	QTextStream write(&file);
+	write << json;
+	file.close();
```

Comment:
You don't need to do this. The file will be automatically closed at the end of this method.